### PR TITLE
Sync automation

### PR DESCRIPTION
**Pull request by Infinite-Echo**
Added github action for updating the master branch without replacing dockerfiles

### CODE REVIEW

**Comment by Infinite-Echo** [comment]
Workflow will automatically merge the upstream formant master branch and then overwrite the dockerfiles and config.json so that they do not get changed by the merge. AKA this will now keep us up to date automatically without changing the necessary parameters for our dockers.

**Comment by Infinite-Echo** [comment]
> 

Also settings for git action permissions will need to be modified in the repo so that the action is able to push code.

File: docker-compose.yaml [hunk 1]
```diff
@@ -14,35 +14,5 @@ services:
     environment:
       ROS_DOMAIN_ID: ${ROS_DOMAIN_ID}
       RMW_IMPLEMENTATION: ${RMW_IMPLEMENTATION}
-    depends_on:
-      - formant-agent
```
Comment (johnjamesmiller) [inline docker-compose.yaml]: @Infinite-Echo  do we need to get rid of this formant-agent piece? our current setup is to run the formant agent in a docker launched from here.

Comment (johnjamesmiller) [reply docker-compose.yaml]: @Infinite-Echo  added back so i can close in this sprint :)

**Review by johnjamesmiller — APPROVED** [review]
good work!

**Pull request merged**